src/main/java/com/synopsys/arc/jenkinsci/plugins/jobrestrictions/util/GroupSelector.java

```diff
@@ -96,7 +97,9 @@ public String getDisplayName() {
             return "N/A";
         }
 
+        @RequirePOST
         public FormValidation doCheckSelectedGroupId(@QueryParameter String selectedGroupId) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Comment:
Afaik users can't find out which groups exist on Jenkins. But the Administer permission is too strict I think.
Maybe add an `@AncestorInPath item` to the checkmethod, this should give you the job you're configuring and then check if you have the `Item.CONFIGURE` permission on that job.